### ISSUE TITLE
Absolute value of velocity missing for topk neurons selection

### ISSUE THREAD

**Issue by aelQuelennec** [EIDOSLAB/NEq#2]
In general_utils.py, in the evaluated_mask function :

```
def evaluated_mask(config, k, reduced_activation_delta, topk, grad_mask):
    if config.eps != "-":
        mask = torch.where(torch.abs(reduced_activation_delta) <= config.eps)[0]
    elif config.binomial:
        mask = torch.where(torch.distributions.binomial.Binomial(probs=reduced_activation_delta).sample() == 0)[0]
    else:
        # How many neurons to select as "to freeze" as percentage of the total number of neurons
        topk = int((1 - topk) * reduced_activation_delta.shape[0])
        mask = torch.topk(reduced_activation_delta, k=topk, largest=False, sorted=True)[1]  
    if config.pinning and k in grad_mask:
        grad_mask[k] = torch.cat([grad_mask[k].long(), mask.long()]).unique()
    else:
        grad_mask[k] =  mask
```

In the epsilon case, the mask corresponds to the neurons having an **absolute** velocity value below a certain threshold.
However, in the  topk case, the neurons are ranked in ascending order with respect to their **real** speed meaning that topk neurons are those with a high negative speed instead of those with a close to 0 absolute speed (which are those we should be freezing).
The neuron selection should be : 

`
mask = torch.topk(torch.abs(reduced_activation_delta), k=topk, largest=False, sorted=True)[1]
`

**Comment by AndreaBrg** [comment]
@aelQuelennec I guess you are working without setting an eps value. Is that right?
In that case, you are correct. The abs operation is missing. That is because that part of the mask evaluation is a leftover from older tests we made. We ended up using only the  `mask = torch.where(torch.abs(reduced_activation_delta) <= config.eps)[0]` part for our final version.

Thanks for pointing that out.

**Comment by aelQuelennec** [comment]
@AndreaBrg Yes, I am currently working with Enzo on a neuron selection such as the number of updated weights remains under a given threshold.

You're welcome !
